File: OCaml/Daily001_easy/UserInput.ml
(*
  user Input 
  Challenge #1 (easy)
  https://www.reddit.com/r/dailyprogrammer/comments/pih8x/easy_challenge_1/ 


  sarthak7u@gmail.com
*)


let main () =
  print_endline "What is your name?";
  let name = read_line () in
  
  print_endline "What is your age?";

  let age = read_line () in

  print_endline "What is your uname?";
  let uname = read_line () in

  let output = "Your name is " ^ name ^ ", your age is " ^ age ^ ", and your username is " ^ uname in 
  print_endline output;
  
  let oc = open_out "output.txt" in
  Printf.fprintf oc "%s" output;
  close_out oc

let () = main ()

File: Swift/Daily001_easy/userInput.swift
/** 
 * User Input 
 *
 * challenge #1 (easy)
 * https://www.reddit.com/r/dailyprogrammer/comments/pih8x/easy_challenge_1/ 
 *
 * sarthak7u@gmail.com
 */

func getField(_ label: String) -> String? {
    print("Your \(label):", terminator: " ")
    let f = readLine()
    return f
}

let name = getField("Name")
let age = getField("Age")
let uname = getField("Username")


if (name != nil && age != nil && uname != nil) {
    print("your name is \(name!), you are \(age!) years old, and your username is \(uname!)")
} else {
    print("Bad Request")
}


